src/flows/pipes/Downsample/index.ts

```diff
@@ -0,0 +1,61 @@
+import View from './view'
+import './style.scss'
+
+import {FUNCTIONS} from 'src/timeMachine/constants/queryBuilder'
+
+export default register => {
+  register({
+    type: 'downsample',
+    family: 'transform',
+    component: View,
+    featureFlag: 'flow-panel--downsample',
+    button: 'Downsample',
+    initial: {
+      functions: [],
+      aggregateWindow: {
+        period: '',
+        fillValues: false,
+      },
+    },
+    generateFlux: (pipe, create) => {
+      if (!pipe.aggregateWindow.period || !pipe.functions.length) {
+        return
+      }
+
+      const _build = (config, fn?): string => {
+        if (config.functions) {
+          return config.functions
+            .map(fnc => {
+              const conf = {...config}
+              delete conf.functions
+              return _build(conf, fnc)
+            })
+            .filter(q => q.length)
+            .join('\n\n')
+        }
+
+        const fnSpec = fn && FUNCTIONS.find(spec => spec.name === fn.name)
+
+        if (!fnSpec) {
+          return ''
+        }
+
+        const period = config.aggregateWindow.period
+        const flux = fnSpec.flux(
+          period === 'auto' || !period ? 'v.windowPeriod' : period,
```

Comment:
wait so this expression either returns `true`, `v.windowPeriod`, or whatever the value of `period` is? Or am I mis-reading it?

Reply:
me rn 
https://media.giphy.com/media/4JVTF9zR9BicshFAb7/giphy.gif

Reply:
lol idk i copy pasta'd it from the other working examples in our code base. read it as: if period is `auto` or falsy, use v.windowPeriod to inherit it from the application state, or else hardcode the period set by the user

Reply:
lol ok. I'm sure it's fine. JS is so weird sometimes